pkg/arvo/sys/arvo.hoon

```diff
@@ -21,65 +21,154 @@
++$  bone  @ud
++$  case
+  $%  ::  %da:  date
+      ::  %tas: label
+      ::  %ud:  sequence
+      ::
+      [%da p=@da]
+      [%tas p=@tas]
+      [%ud p=@ud]
+  ==
++$  cage  (cask vase)
+++  cask  |$  [a]  (pair mark a)
++$  desk  @tas
++$  dock  (pair @p term)
++$  mark  @tas
++$  ship  @p
++$  sink  (trel bone ship path)
+::
++|  %meta
+::
+::  +hypo: type-associated builder
+::  $meta: meta-vase
+::  $maze: vase, or meta-vase
+::
+++  hypo
+  |$  [a]
+  (pair type a)
++$  meta  (pair)
++$  maze  (each vase meta)
+::
++|  %interface
+::
+::  $ball: dynamic kernel action
+::  $curd: tagged, untyped event
+::  $duct: causal history
+::  +hobo: %soft task builder
+::  $goof: crash label and trace XX fail/ruin/crud/flaw/lack/miss
+::  $mass: memory usage
+::  $monk: general identity
+::  $move: cause and action
+::  $ovum: card with cause
+::  $scry-sample: vane +scry argument
+::  $vane-sample: vane wrapper-gate aargument
+::  $sley: namespace function
+::  $slyd: super advanced
+::  $slyt: old namespace
+::  +wind: kernel action builder
+::  $wire: event pretext
+::  +wite: kernel action/error builder
+::
++$  ball  (wite [vane=term task=maze] maze)
++$  curd  (cask)
++$  duct  (list wire)
+++  hobo
+  |$  [a]
+  $?  $%  [%soft p=*]
+      ==
+      a
+  ==
++$  goof  [mote=term =tang]
++$  mass  $~  $+|+~
+          (pair cord (each * (list mass)))
++$  monk  (each ship (pair @tas @ta))
++$  move  [=duct =ball]
++$  ovum  (pair wire curd)
+::
 +$  scry-sample
   [fur=(unit (set monk)) ren=@tas why=shop syd=desk lot=coin tyl=path]
 +$  vane-sample
   [our=ship now=@da eny=@uvJ ski=slyd]
-++  ship  @p                                            ::  network identity
-++  sink  (trel bone ship path)                         ::  subscription
-++  sley  $-  {* (unit (set monk)) term beam}           ::  namespace function
-          (unit (unit cage))                            ::
-++  slyd  $-  {* (unit (set monk)) term beam}           ::  super advanced
-          (unit (unit (cask milt)))                     ::
-++  slyt  $-({* *} (unit (unit)))                       ::  old namespace
+::
++$  sley
+  $-  [* (unit (set monk)) term beam]
+  (unit (unit cage))
++$  slyd
+  $~  =>(~ |~(* ~))
+  $-  [* (unit (set monk)) term beam]
+  (unit (unit (cask meta)))
++$  slyt  $-(^ (unit (unit)))
+::
+++  wind
+  |$  ::  a: forward
+      ::  b: reverse
+      ::
+      [a b]
+  $%  ::  %pass: advance
+      ::  %slip: lateral
+      ::  %give: retreat
+      ::
+      [%pass p=path q=a]
+      [%slip p=a]
+      [%give p=b]
+  ==
++$  wire  path
+++  wite
+  |$  ::  note: a routed $task
+      ::  gift: a reverse action
+      ::
+      ::    NB:  task: a forward action
+      ::         sign: a sourced $gift
+      ::
+      [note gift]
+  $%  ::  %hurl: action failed
+      ::  %pass: advance
+      ::  %slip: lateral
+      ::  %give: retreat
+      ::
+      [%hurl =goof wite=$;($>(?(%pass %give) $))]
+      [%pass =wire =note]
+      [%slip =note]
```

Comment:
I see you decided to leave %slip in there.  Do we still use it anywhere?  Also, what happens if a %slip fails, since it can't get wrapped in a %hurl?

Reply:
We `%slip %init` to all the vanes during the "legacy boot" event. And depend on the resulting duct(s), IIRC. We're not gonna recover from a failure in that event. Sane vane bootstrap is definitely on my list, but it could get messy (and might not be backwards compatible), so I'm saving it for later.